.github/workflows/python-test.yml

```diff
@@ -28,31 +28,291 @@ jobs:
+        # install lgb server dependencies if cache does not exist
+      - name: Install lgb dependencies
+        if: steps.cached-lgb-dependencies.outputs.cache-hit != 'true'
+        run: |
+          cd python/lgbserver
+          poetry install --with=test --no-interaction --no-root
+      - name: Install lgbserver
+        run: |
+          cd python/lgbserver
+          poetry install --no-interaction
+      - name: Test lgbserver
+        run: |
+          cd python
+          source lgbserver/.venv/bin/activate
           pytest --cov=lgbserver ./lgbserver
-          pip install -e ./paddleserver[test]
+
+      # ----------------------------------------Paddle Server Unit Tests------------------------------------------------
+      # load cached paddle venv if cache exists
+      - name: Load cached paddle venv
+        id: cached-paddle-dependencies
+        uses: actions/cache@v3
+        with:
+          path: python/paddleserver/.venv
+          key: paddle-venv-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/kserve/poetry.lock, **/paddleserver/poetry.lock') }}
+        # install paddle server dependencies if cache does not exist
+      - name: Install paddle dependencies
+        if: steps.cached-paddle-dependencies.outputs.cache-hit != 'true'
+        run: |
+          cd python/paddleserver
+          poetry install --with=test --no-interaction --no-root
+      - name: Install paddleserver
+        run: |
+          cd python/paddleserver
+          poetry install --no-interaction
+      - name: Test paddleserver
+        run: |
+          cd python
+          source paddleserver/.venv/bin/activate
           pytest --cov=paddleserver ./paddleserver
-          pip install -e ./alibiexplainer[test]
+
+      # ----------------------------------------Alibi Explainer Unit Tests------------------------------------------------
+      # load cached alibi venv if cache exists
+      - name: Load cached alibi venv
+        id: cached-alibi-dependencies
+        uses: actions/cache@v3
+        with:
+          path: python/alibiexplainer/.venv
+          key: alibi-venv-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/kserve/poetry.lock, **/alibiexplainer/poetry.lock') }}
+        # install alibi explainer dependencies if cache does not exist
+      - name: Install alibi dependencies
+        if: steps.cached-alibi-dependencies.outputs.cache-hit != 'true'
+        run: |
+          cd python/alibiexplainer
+          poetry install --with=test --no-interaction --no-root
+      - name: Install alibiexplainer
+        run: |
+          cd python/alibiexplainer
+          poetry install --no-interaction
+      - name: Test alibiexplainer
+        run: |
+          cd python
+          source alibiexplainer/.venv/bin/activate
           pytest --cov=alibiexplainer ./alibiexplainer
+  build-3-7:
```

Comment:
Is there anything special handling for python 3.7 ? though this is going to be EOL soon in 6/2023
https://devguide.python.org/versions/

Reply:
Some of them, had requirements set as above 3.7 in their setup.py, though they were running in a matrix of 3.7, 3.8 and 3.9. So, it was segregated to make it explicit.
As you said, we can remove this once we remove support for 3.7, in the near future.